plugins/workflow/category/src/Extension/Category.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Workflow.category
+ *
+ * @copyright   (C) 2025 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Plugin\Workflow\Category\Extension;
+
+use Joomla\CMS\Event\Model\PrepareFormEvent;
+use Joomla\CMS\Event\Workflow\WorkflowTransitionEvent;
+use Joomla\CMS\Event\View\DisplayEvent;
+use Joomla\CMS\Form\Form;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Workflow\WorkflowPluginTrait;
+use Joomla\Database\DatabaseAwareTrait;
+use Joomla\Event\SubscriberInterface;
+use Joomla\Registry\Registry;
+
+// phpcs:disable PSR1.Files.SideEffects
+defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+/**
+ * Workflow Category Transition Plugin
+ *
+ * @since  DEPLOY_VERSION
+ */
+final class Category extends CMSPlugin implements SubscriberInterface
+{
+    use WorkflowPluginTrait;
+    use DatabaseAwareTrait;
+    /**
+     * Load the language file on instantiation.
+     *
+     * @var    bool
+     * @since  DEPLOY_VERSION
+     */
+    protected $autoloadLanguage = true;
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'onAfterDisplay'             => 'onAfterDisplay',
+            'onContentPrepareForm'       => 'onContentPrepareForm',
+            'onWorkflowAfterTransition'  => 'onWorkflowAfterTransition',
+        ];
+    }
+
+    /**
+     * The form event.
+     *
+     * @param   PrepareFormEvent  $event  The event
+     *
+     * @since   DEPLOY_VERSION
+     */
+    public function onContentPrepareForm(PrepareFormEvent $event)
+    {
+        $form = $event->getForm();
+        $data = $event->getData();
+
+        $context = $form->getName();
+
+        // Extend the transition form
+        if ($context === 'com_workflow.transition') {
+            $this->enhanceWorkflowTransitionForm($form, $data);
+
+            return;
+        }
+
+        if ($context === 'com_content.article') {
```

Comment:
Here we should think about a way to not limit to com_content only (check the featuring plugin, how what options there are)